### ISSUE TITLE
Forum displays an empty page when spam entries should be displayed

### ISSUE THREAD

**Issue by auge8472** [ilosuna/mylittleforum#621]
When one is *not* on page 1 of the main view (threads listing) and wants to check entries that was classified as spam, it is very likely to see an empty page with the message "No messages have been posted so far.". This depends on the amount of existing entries that got classified as spam but in most cases there will be not enough entries to fill more than one page.

This is misleading, because the forum tells me, that there are spam entries in the admin options but the message tells me "There are none" on the other hand. I assume, that the forum keeps the information about the currently selected page of the main view and redirects me to there, when I requested the list of spam entries, even there are not enough spam entries to show them on the selected page number.

**Comment by loesler** [comment]
You are right. The reason is the wrong page stored in `$_SESSION[$settings['session_prefix'].'usersettings']['page']`. So, a simple solution is to reset the value around lines [192ff](https://github.com/ilosuna/mylittleforum/blob/master/includes/main.inc.php#L192) by


```
if (isset($_GET['show_spam']) && isset($_SESSION[$settings['session_prefix'].'user_id']) && $_SESSION[$settings['session_prefix'].'user_id'] > 0) {
	if (isset($_SESSION[$settings['session_prefix'].'usersettings']['show_spam'])) unset($_SESSION[$settings['session_prefix'].'usersettings']['show_spam']);
	else $_SESSION[$settings['session_prefix'].'usersettings']['show_spam'] = true;
	$_SESSION[$settings['session_prefix'].'usersettings']['page'] = 1; // reset page number
	header('location: index.php?mode=index');
	exit;
}
```

The limitation is, however, if you switch back, you are on the first page. It is not perfect but I think it is okay.
